File: backend/WMS-mini/src/WMS.Domain/Entities/Warehouse.cs
using System.ComponentModel.DataAnnotations;
using WMS.Domain.Common;

namespace WMS.Domain.Entities;

public class Warehouse : BaseAuditableEntity
{
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;

    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Address { get; set; }

    public ICollection<Location> Locations { get; set; } = new List<Location>();
}

File: frontend/src/types/warehouse.ts
export interface WarehouseDto {
  id: string
  code: string
  name: string
  address?: string
}

export interface CreateWarehouseDto {
  code: string
  name: string
  address?: string
}

// Backend UpdateWarehouseDto chỉ nhận name + address (code không thể sửa)
export interface UpdateWarehouseDto {
  name: string
  address?: string
}